tensorboard/data/server/run.rs

```diff
@@ -330,28 +394,17 @@ mod test {
                 ..Default::default()
             }
         );
-        let mut basin = crate::reservoir::Basin::new();
-        ts.rsv.commit(&mut basin);
 
         // Points should be as expected (no downsampling at these sizes).
-        let mut actual_points = Vec::new();
-        for (step, StageValue { wall_time, payload }) in basin.as_slice() {
-            if let EventValue::Summary(SummaryValue(value_box)) = payload {
-                if let pb::summary::value::Value::SimpleValue(f) = **value_box {
-                    actual_points.push((*step, *wall_time, f));
-                    continue;
-                }
-            }
-            panic!("step {:?}: {:?}", step, payload);
-        }
+        let scalar = |f: f32| commit::ScalarValue(f64::from(f));
```

Comment:
Yeah—[Clippy nagged me about this once][1], so I’ve been preferring
`from` since. Clippy [doesn’t seem to complain on that code anymore][2],
though, so maybe they removed or changed that lint? I don’t personally
have a strong preference, thuogh my sense is that there’s a consensus
that `as`-conversions are considered a bit of a wart.

[1]: https://github.com/wchargin/rustboard/commit/a48be2ad0b0b4e933fc3468414ae2d30a0bc6321
[2]: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=29cf41e7fe5561161ac54f3dac96dbea